Have a better Exception message for "Invalid URL host: null"

Full Exception:

```java
Exception: tv.hd3g.commons.IORuntimeException: Invalid URL host: "null"
	at tv.hd3g.transfertfiles.AbstractFileSystemURL.<init>(AbstractFileSystemURL.java:128)
	at tv.hd3g.jobkit.watchfolder.ObservedFolder.createFileSystem(ObservedFolder.java:117)
	at tv.hd3g.jobkit.watchfolder.Watchfolders.internalScan(Watchfolders.java:93)
	at tv.hd3g.jobkit.watchfolder.Watchfolders.lambda$startScans$6(Watchfolders.java:129)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:720)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at tv.hd3g.jobkit.watchfolder.Watchfolders.lambda$startScans$7(Watchfolders.java:138)
	at tv.hd3g.jobkit.engine.SpoolExecutor$SpoolJob.run(SpoolExecutor.java:146)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.net.UnknownHostException: <FQDN host>: Nom ou service inconnu
	at java.base/java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method)
	at java.base/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:929)
	at java.base/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1515)
	at java.base/java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:848)
	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1505)
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1364)
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1298)
	at java.base/java.net.InetAddress.getByName(InetAddress.java:1248)
	at tv.hd3g.transfertfiles.AbstractFileSystemURL.<init>(AbstractFileSystemURL.java:76)
	... 14 more
```

--- a/pom.xml
+++ b/pom.xml
@@ -3,7 +3,7 @@
 
     <groupId>tv.hd3g</groupId>
     <artifactId>transfertfiles</artifactId>
-    <version>0.2.1</version>
+    <version>0.3.0-SNAPSHOT</version>
     <packaging>jar</packaging>
 
     <name>transfertfiles</name>

--- a/src/main/java/tv/hd3g/transfertfiles/AbstractFileSystemURL.java
+++ b/src/main/java/tv/hd3g/transfertfiles/AbstractFileSystemURL.java
@@ -68,65 +68,69 @@ public class AbstractFileSystemURL implements Closeable {
 	 *        Add "timeout=5" for set 5 seconds of connect/socket timeout
 	 */
 	public AbstractFileSystemURL(final String ressourceURL) {
-		try {
-			final var url = new URLAccess(ressourceURL);
+		final var url = new URLAccess(ressourceURL);
 
-			final var protocol = url.getProtocol();
-			basePath = normalizePath(Optional.ofNullable(url.getPath()).orElse("/"));
-			final var host = InetAddress.getByName(Optional.ofNullable(url.getHost()).orElse("localhost"));
-			final var query = url.getOptionZone();
-			final var username = url.getUsername();
-			protectedRessourceURL = url.getProtectedRessourceURL();
-			final var password = url.getPassword();
-			final var port = url.getPort();
+		final var protocol = url.getProtocol();
+		basePath = normalizePath(Optional.ofNullable(url.getPath()).orElse("/"));
+		final var host = resolveHostname(url);
+		final var query = url.getOptionZone();
+		final var username = url.getUsername();
+		protectedRessourceURL = url.getProtectedRessourceURL();
+		final var password = url.getPassword();
+		final var port = url.getPort();
 
-			final var passive = query.containsKey("active") == false;
-			final var ignoreInvalidCertificates = query.containsKey("ignoreInvalidCertificates");
-			final var keys = query.getOrDefault("key", List.of()).stream()
-			        .map(File::new)
-			        .filter(File::exists)
-			        .collect(toUnmodifiableList());
+		final var passive = query.containsKey("active") == false;
+		final var ignoreInvalidCertificates = query.containsKey("ignoreInvalidCertificates");
+		final var keys = query.getOrDefault("key", List.of()).stream()
+		        .map(File::new)
+		        .filter(File::exists)
+		        .collect(toUnmodifiableList());
 
-			if (protocol.contentEquals("file")) {
-				log.debug("Init URL LocalFileSystem: {}", this::toString);
-				fileSystem = new LocalFileSystem(new File(basePath));
-			} else if (protocol.contentEquals("sftp")) {
-				log.debug("Init URL SFTPFileSystem: {}", this::toString);
-				fileSystem = new SFTPFileSystem(host, port, username, basePath);
-				final var sFTPfileSystem = (SFTPFileSystem) fileSystem;
+		if (protocol.contentEquals("file")) {
+			log.debug("Init URL LocalFileSystem: {}", this::toString);
+			fileSystem = new LocalFileSystem(new File(basePath));
+		} else if (protocol.contentEquals("sftp")) {
+			log.debug("Init URL SFTPFileSystem: {}", this::toString);
+			fileSystem = new SFTPFileSystem(host, port, username, basePath);
+			final var sFTPfileSystem = (SFTPFileSystem) fileSystem;
 
-				if (keys.isEmpty()) {
-					if (password.length > 0) {
-						sFTPfileSystem.setPasswordAuth(password);
-					}
-				} else {
-					keys.forEach(privateKey -> sFTPfileSystem.manuallyAddPrivatekeyAuth(privateKey, password));
+			if (keys.isEmpty()) {
+				if (password.length > 0) {
+					sFTPfileSystem.setPasswordAuth(password);
 				}
-			} else if (protocol.contentEquals("ftp")) {
-				log.debug("Init URL FTPFileSystem: {}", this::toString);
-				fileSystem = new FTPFileSystem(host, port, username, password, passive, basePath);
-			} else if (protocol.contentEquals("ftps")) {
-				log.debug("Init URL FTPSFileSystem: {}", this::toString);
-				fileSystem = new FTPSFileSystem(host, port, username, password, passive,
-				        ignoreInvalidCertificates, basePath);
-			} else if (protocol.contentEquals("ftpes")) {
-				log.debug("Init URL FTPESFileSystem: {}", this::toString);
-				fileSystem = new FTPESFileSystem(host, port, username, password, passive,
-				        ignoreInvalidCertificates, basePath);
 			} else {
-				throw new UncheckedIOException(
-				        new IOException("Can't manage protocol \"" + protocol + "\" in URL: " + toString()));
+				keys.forEach(privateKey -> sFTPfileSystem.manuallyAddPrivatekeyAuth(privateKey, password));
 			}
+		} else if (protocol.contentEquals("ftp")) {
+			log.debug("Init URL FTPFileSystem: {}", this::toString);
+			fileSystem = new FTPFileSystem(host, port, username, password, passive, basePath);
+		} else if (protocol.contentEquals("ftps")) {
+			log.debug("Init URL FTPSFileSystem: {}", this::toString);
+			fileSystem = new FTPSFileSystem(host, port, username, password, passive,
+			        ignoreInvalidCertificates, basePath);
+		} else if (protocol.contentEquals("ftpes")) {
+			log.debug("Init URL FTPESFileSystem: {}", this::toString);
+			fileSystem = new FTPESFileSystem(host, port, username, password, passive,
+			        ignoreInvalidCertificates, basePath);
+		} else {
+			throw new UncheckedIOException(
+			        new IOException("Can't manage protocol \"" + protocol + "\" in URL: " + toString()));
+		}
 
-			final var timeout = query.getOrDefault("timeout", List.of()).stream()
-			        .map(Integer::valueOf)
-			        .findFirst()
-			        .orElse(0);
-			if (timeout > 0) {
-				fileSystem.setTimeout(timeout, SECONDS);
-			}
+		final var timeout = query.getOrDefault("timeout", List.of()).stream()
+		        .map(Integer::valueOf)
+		        .findFirst()
+		        .orElse(0);
+		if (timeout > 0) {
+			fileSystem.setTimeout(timeout, SECONDS);
+		}
+	}
+
+	private InetAddress resolveHostname(final URLAccess url) {
+		try {
+			return InetAddress.getByName(Optional.ofNullable(url.getHost()).orElse("localhost"));
 		} catch (final UnknownHostException e) {
-			throw new UncheckedIOException("Invalid URL host: \"" + toString() + "\"", e);
+			throw new UncheckedIOException("Can't resolve hostname: \"" + url.getHost() + "\"", e);
 		}
 	}
 

--- a/src/test/java/tv/hd3g/transfertfiles/AbstractFileSystemURLTest.java
+++ b/src/test/java/tv/hd3g/transfertfiles/AbstractFileSystemURLTest.java
@@ -21,6 +21,7 @@ import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
@@ -28,6 +29,7 @@ import static org.mockito.Mockito.when;
 
 import java.io.File;
 import java.io.IOException;
+import java.io.UncheckedIOException;
 import java.net.UnknownHostException;
 
 import org.junit.jupiter.api.AfterEach;
@@ -152,6 +154,17 @@ class AbstractFileSystemURLTest {
 	void testSetInvalidTimeout() {
 		final var url = "file://localhost/" + new File("").getAbsolutePath() + "?timeout=NOPE";
 		assertThrows(NumberFormatException.class, () -> new AbstractFileSystemURL(url));
+	}
+
+	@Test
+	void testBadResolve() throws IOException {
+		final var url = "ftpes://this-host-dont-exists/";
+		try (var afs = new AbstractFileSystemURL(url)) {
+			fail("Expect exception");
+		} catch (final UncheckedIOException e) {
+			assertTrue(e.getCause().getClass().isAssignableFrom(UnknownHostException.class));
+			assertEquals("Can't resolve hostname: \"this-host-dont-exists\"", e.getMessage());
+		}
 	}
 
 	static class TestAbstractFileSystemURL extends AbstractFileSystemURL {

--- a/src/test/java/tv/hd3g/transfertfiles/AbstractFileSystemURLTest.java
+++ b/src/test/java/tv/hd3g/transfertfiles/AbstractFileSystemURLTest.java
@@ -22,7 +22,6 @@ import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.fail;
-import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
@@ -203,12 +202,12 @@ class AbstractFileSystemURLTest {
 		void testGetFromPath() {
 			final var path = String.valueOf(System.nanoTime());
 			final var fullPath = "/" + path;
-			when(fileSystem.getFromPath(eq(fullPath))).thenReturn(f);
+			when(fileSystem.getFromPath(fullPath)).thenReturn(f);
 			final var p = afs.getFromPath(path);
 			assertNotNull(p);
 			assertEquals(f, p);
 			verify(fileSystem, times(1)).connect();
-			verify(fileSystem, times(1)).getFromPath(eq(fullPath));
+			verify(fileSystem, times(1)).getFromPath(fullPath);
 		}
 
 		@Test
@@ -218,12 +217,12 @@ class AbstractFileSystemURLTest {
 
 		@Test
 		void testGetRootPath() {
-			when(fileSystem.getFromPath(eq("/"))).thenReturn(f);
+			when(fileSystem.getFromPath("/")).thenReturn(f);
 			final var r = afs.getRootPath();
 			assertNotNull(r);
 			assertEquals(f, r);
 			verify(fileSystem, times(1)).connect();
-			verify(fileSystem, times(1)).getFromPath(eq("/"));
+			verify(fileSystem, times(1)).getFromPath("/");
 		}
 
 		@Test

--- a/src/test/java/tv/hd3g/transfertfiles/CommonAbstractFileTest.java
+++ b/src/test/java/tv/hd3g/transfertfiles/CommonAbstractFileTest.java
@@ -21,7 +21,6 @@ import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.ArgumentMatchers.anyLong;
-import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.atLeastOnce;
 import static org.mockito.Mockito.only;
 import static org.mockito.Mockito.verify;
@@ -83,9 +82,9 @@ class CommonAbstractFileTest {
 
 	@Test
 	void testGetParent() {
-		Mockito.when(fs.getFromPath(eq(parentPath))).thenReturn(parent);
+		Mockito.when(fs.getFromPath(parentPath)).thenReturn(parent);
 		assertEquals(parent, caf.getParent());
-		Mockito.verify(fs, only()).getFromPath(eq(parentPath));
+		Mockito.verify(fs, only()).getFromPath(parentPath);
 	}
 
 	@Test
@@ -156,9 +155,9 @@ class CommonAbstractFileTest {
 		final var totalSize = (int) observableCopyStream(inputStream, outputStream, 10, copyCallback);
 
 		assertEquals(30, totalSize);
-		verify(copyCallback, Mockito.times(1)).apply(eq(10L));
-		verify(copyCallback, Mockito.times(1)).apply(eq(20L));
-		verify(copyCallback, Mockito.times(1)).apply(eq(30L));
+		verify(copyCallback, Mockito.times(1)).apply(10L);
+		verify(copyCallback, Mockito.times(1)).apply(20L);
+		verify(copyCallback, Mockito.times(1)).apply(30L);
 		verifyNoMoreInteractions(copyCallback);
 
 		assertTrue(Arrays.equals(sourceDatas, 0, totalSize, outputStream.toByteArray(), 0, totalSize));

--- a/src/test/java/tv/hd3g/transfertfiles/ftp/FTPListingTest.java
+++ b/src/test/java/tv/hd3g/transfertfiles/ftp/FTPListingTest.java
@@ -17,7 +17,6 @@
 package tv.hd3g.transfertfiles.ftp;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.mockito.ArgumentMatchers.eq;
 
 import java.io.IOException;
 
@@ -59,11 +58,11 @@ class FTPListingTest {
 		var stream = FTPListing.NLST.listDirectory(client, path);
 		assertEquals(0, stream.count());
 
-		Mockito.when(client.listNames(eq(path))).thenReturn(new String[] { result });
+		Mockito.when(client.listNames(path)).thenReturn(new String[] { result });
 		stream = FTPListing.NLST.listDirectory(client, path);
 		assertEquals(result, stream.findFirst().get());
 
-		Mockito.verify(client, Mockito.times(2)).listNames(eq(path));
+		Mockito.verify(client, Mockito.times(2)).listNames(path);
 	}
 
 	@Test
@@ -71,11 +70,11 @@ class FTPListingTest {
 		var stream = FTPListing.LIST.listDirectory(client, path);
 		assertEquals(0, stream.count());
 
-		Mockito.when(client.listFiles(eq(path))).thenReturn(new org.apache.commons.net.ftp.FTPFile[] { fileEntry });
+		Mockito.when(client.listFiles(path)).thenReturn(new org.apache.commons.net.ftp.FTPFile[] { fileEntry });
 		stream = FTPListing.LIST.listDirectory(client, path);
 		assertEquals(result, stream.findFirst().get());
 
-		Mockito.verify(client, Mockito.times(2)).listFiles(eq(path));
+		Mockito.verify(client, Mockito.times(2)).listFiles(path);
 		Mockito.verify(fileEntry, Mockito.times(2)).getName();
 	}
 
@@ -84,11 +83,11 @@ class FTPListingTest {
 		var stream = FTPListing.MLSD.listDirectory(client, path);
 		assertEquals(0, stream.count());
 
-		Mockito.when(client.mlistDir(eq(path))).thenReturn(new org.apache.commons.net.ftp.FTPFile[] { fileEntry });
+		Mockito.when(client.mlistDir(path)).thenReturn(new org.apache.commons.net.ftp.FTPFile[] { fileEntry });
 		stream = FTPListing.MLSD.listDirectory(client, path);
 		assertEquals(result, stream.findFirst().get());
 
-		Mockito.verify(client, Mockito.times(2)).mlistDir(eq(path));
+		Mockito.verify(client, Mockito.times(2)).mlistDir(path);
 		Mockito.verify(fileEntry, Mockito.times(2)).getName();
 	}
 

--- a/src/test/java/tv/hd3g/transfertfiles/ftp/StoppableOutputStreamTest.java
+++ b/src/test/java/tv/hd3g/transfertfiles/ftp/StoppableOutputStreamTest.java
@@ -19,7 +19,6 @@ package tv.hd3g.transfertfiles.ftp;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.times;
 
 import java.io.IOException;
@@ -53,7 +52,7 @@ class StoppableOutputStreamTest {
 	@Test
 	void testWriteInt() throws IOException {
 		os.write(42);
-		Mockito.verify(out, Mockito.only()).write(eq(42));
+		Mockito.verify(out, Mockito.only()).write(42);
 		os.setStop();
 		assertThrows(IOException.class, () -> os.write(42));
 		Mockito.verify(out, times(1)).flush();
@@ -64,7 +63,7 @@ class StoppableOutputStreamTest {
 	void testWriteByteArray() throws IOException {
 		final var v = new byte[] { 11 };
 		os.write(v);
-		Mockito.verify(out, Mockito.only()).write(eq(11));
+		Mockito.verify(out, Mockito.only()).write(11);
 		os.setStop();
 		assertThrows(IOException.class, () -> os.write(v));
 		Mockito.verify(out, times(1)).flush();
@@ -75,7 +74,7 @@ class StoppableOutputStreamTest {
 	void testWriteByteArrayIntInt() throws IOException {
 		final var v = new byte[] { 11 };
 		os.write(v, 0, 1);
-		Mockito.verify(out, Mockito.only()).write(eq(11));
+		Mockito.verify(out, Mockito.only()).write(11);
 		os.setStop();
 		assertThrows(IOException.class, () -> os.write(v, 0, 1));
 		Mockito.verify(out, times(1)).flush();

